### ISSUE TITLE
DatasetUtils.findItemIndicesForX has a small bug.

### ISSUE THREAD

**Issue by claudio-rosati** [jfree/jfreechart#273]
DatasetUtils.findItemIndicesForX has a small bug: at line 2286, before closing the `for` block the following missing statement must be added:
``` Java
prev = next;
```

**Comment by jfree** [comment]
You are right, but I also cannot create a failing JUnit test.  Do you have a failing case?

**Comment by claudio-rosati** [comment]
If you don't fix it, the first point of a series will be used for the interpolation. So if you use a series with 3 point, where 1 and 2 define a horizontal segment, while 2 and 3 an almost vertical one, interpolating between 2 and 3 should give a substantial difference.
But I've not tried this. I was creating 2 utility methods to give all vertical intersections and all horizontal intersections using your code as starting point. Doing so I've discovered the problem, just looking at the code, not because data where wrong.

**Comment by jfree** [comment]
Thanks for coming back on that.  I think the result is OK because of the return value being:

`return new int[] {i - 1, i}; // spanning match`

That said, I will fix the code because it doesn't read well 😃
